# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1219788313 14400
#      Tue Aug 26 18:05:13 2008 -0400
# Branch release-3-0-x
# Node ID 965bb17961f060622e607c6eaeaba1f813aede3d
# Parent  8f28953882c25efddc38d11609fd392781bfb205
hold.m: if hold is applied to a figure, set state for all child axes objects

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-08-26  John W. Eaton  <jwe@octave.org>
+
+	* plot/hold.m: If hold is applied to a figure, set state for all
+	child axes objects.
+
 2008-08-22  John W. Eaton  <jwe@octave.org>
 
 	* statistics/distributions/chi2rnd.m: Fix missing semicolon.
 
 2008-08-21  John W. Eaton  <jwe@octave.org>
 
 	* plot/__go_draw_axes__.m: Fix test for have_data.
 	(__gnuplot_write_data__): Write "Inf Inf\n" if all data pairs
diff --git a/scripts/plot/hold.m b/scripts/plot/hold.m
--- a/scripts/plot/hold.m
+++ b/scripts/plot/hold.m
@@ -33,22 +33,25 @@
 ## state off, and @code{hold} with no arguments toggles the current hold
 ## state.
 ## @end deftypefn
 
 ## PKG_ADD: mark_as_command hold
 
 function hold (varargin)
 
-  [h, varargin] = __plt_get_axis_arg__ ("hold", varargin{:});
+  if (nargin > 0 && ishandle (varargin{1}))
+    [h, varargin, nargs] = __plt_get_axis_arg__ ("hold", varargin{:});
+  else
+    h = gcf ();
+    nargs = numel (varargin);
+  endif
 
   hold_state = get (h, "nextplot");
 
-  nargs = numel (varargin);
-
   if (nargs == 0)
     if (strcmp (hold_state, "add"))
       hold_state = "replace";
     else
       hold_state = "add";
     endif
   elseif (nargs == 1)
     state = varargin{1};
@@ -60,11 +63,16 @@ function hold (varargin)
       else
 	print_usage ();
       endif
     endif
   else
     print_usage ();
   endif
 
+  if (isfigure (h))
+    axes_objs = findobj (h, "type", "axes");
+    h = [h; axes_objs];
+  endif
+
   set (h, "nextplot", hold_state);
 
 endfunction
